localstack-core/localstack/services/apigateway/next_gen/execute_api/helpers.py

```diff
@@ -113,3 +115,27 @@ def validate_sub_dict_of_typed_dict(typed_dict: Type[TypedDict], obj: dict) -> b
     typed_dict_keys = {*typed_dict.__required_keys__, *typed_dict.__optional_keys__}
 
     return not bool(set(obj) - typed_dict_keys)
+
+
+def generate_trace_id():
+    """https://docs.aws.amazon.com/xray/latest/devguide/xray-api-sendingdata.html#xray-api-traceids"""
+    original_request_epoch = int(time.time())
+    timestamp_hex = hex(original_request_epoch)[2:]
+    version_number = "1"
+    unique_id = token_hex(12)
+    return f"{version_number}-{timestamp_hex}-{unique_id}"
+
+
+def generate_trace_parent():
+    return token_hex(8)
+
+
+def parse_trace_id(trace_id: str) -> dict[str, str]:
+    split_trace = trace_id.split(";")
+    trace_values = {}
+    for trace_part in split_trace:
+        key_value = trace_part.split("=")
+        if len(key_value) == 2:
+            trace_values[key_value[0].capitalize()] = key_value[1]
+
+    return trace_values
```

Comment:
Question: Do you think we should put those method in a shared utils for more services to have access to? I feel like they will mostly be the same for everywhere?

Reply:
That's a good point! Not sure where it should live for now 🤔 we might keep it here for now, and if we hear about more x-ray traces, move it somewhere else?

I believe lambda might have a similar implementation..

Reply:
I think Lambda is only using `long_uid()`, unless I missed it. 

I was thinking that it could make sense in `localstack-core/utils` maybe create a `trace.py` but not super convinced either.

I think it is fine with it here for now, but we should stay alert to prevent this behavior to be implemented many times in the code base!